prow/metrics/http_test.go

```diff
@@ -56,3 +65,192 @@ func TestPowersOfTwoBetween(t *testing.T) {
 		})
 	}
 }
+
+func TestWriteHeader(t *testing.T) {
+	testcases := []struct {
+		name       string
+		statusCode int
+	}{
+		{
+			"StatusOK",
+			http.StatusOK,
+		},
+		{
+			"StatusNotFound",
+			http.StatusNotFound,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			trw := &traceResponseWriter{ResponseWriter: rr, statusCode: http.StatusOK}
+			trw.WriteHeader(tc.statusCode)
+			if rr.Code != tc.statusCode {
+				t.Errorf("mismatch in response headers: expected %s, got %s", http.StatusText(tc.statusCode), http.StatusText(rr.Code))
+			}
+			if trw.statusCode != tc.statusCode {
+				t.Errorf("mismatch in TraceResponseWriter headers: expected %s, got %s", http.StatusText(tc.statusCode), http.StatusText(trw.statusCode))
+			}
+		})
+
+	}
+
+}
+
+func TestWrite(t *testing.T) {
+	testcases := []struct {
+		name         string
+		responseBody string
+	}{
+		{
+			"SimpleText",
+			"Simple text to test response size",
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			trw := &traceResponseWriter{ResponseWriter: rr, statusCode: http.StatusOK}
+			resp := []byte(tc.responseBody)
+			_, err := trw.Write(resp)
+			if err != nil {
+				t.Fatalf("failed to write to TraceResponseWriter")
+			}
+			if rr.Body.String() != tc.responseBody {
+				t.Errorf("mismatch in response body: expected %s, got %s", tc.responseBody, rr.Body.String())
+			}
+			if trw.size != len(resp) {
+				t.Errorf("mismatch in TraceResponseWriter size: expected %d, got %d", len(resp), trw.size)
+			}
+		})
+	}
+}
+
+func TestRecordError(t *testing.T) {
+	var recordErrorCalls = 0
+	testcases := []struct {
+		name          string
+		namespace     string
+		expectedError string
+		expectedCount int
+		expectedOut   string
+	}{
+		{
+			name:          "Simple Error String",
+			namespace:     "testnamespace",
+			expectedError: "sample error message to ensure proper working",
+			expectedOut: `# HELP testnamespace_error_rate number of errors, sorted by label/type
+					   # TYPE testnamespace_error_rate counter
+					   testnamespace_error_rate{error="sample error message to ensure proper working"} %d
+					   `,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			errorRate := ErrorRate(tc.namespace)
+			RecordError(tc.expectedError, errorRate)
+			recordErrorCalls++
```

Comment:
Same here, this must always be one and happens to work?

Reply:
This was helpful when the metrics was a global variable in the test, its no longer relevant. I'll change that